okhttp/src/main/java/io/grpc/okhttp/OkHttpWritableBufferAllocator.java

```diff
@@ -27,9 +27,6 @@
  */
 class OkHttpWritableBufferAllocator implements WritableBufferAllocator {
 
-  // Use 4k as our minimum buffer size.
-  private static final int MIN_BUFFER = 4096;
```

Comment:
Okio Buffer works quite differently than Netty. Okio uses Segments that are all the same size and then combines adjacent buffers to avoid excessive internal fragmentation. We should leave this in-place and actually set [it to 8192](https://github.com/square/okio/blob/d169b4387c229704710dda4b48e8767937a6dd33/okio/src/commonMain/kotlin/okio/Segment.kt#L181).

Reply:
(If we were more serious, we'd round up: `(capacityHint + 8191) / 8192 * 8192`.)